Fix: Search Not Working for 'Name' Column in Admin Standard License Comments

sakshammishra112 opened the pull request:
<!-- SPDX-FileCopyrightText: © Fossology contributors

     SPDX-License-Identifier: GPL-2.0-only
-->

<!-- Please refer to CONTRIBUTING.md (https://github.com/fossology/fossology/blob/master/CONTRIBUTING.md)
before creating the pull request to make sure you follow all the standards. -->

## Description

In the **Admin Standard License Comments** section, searching by the 'Name' column was not functioning correctly. The search only worked for the Comment column but ignored the Name input field. This issue occurred because DataTables was trying to search using the full HTML string of the input field instead of its actual value.The column is easily searchable because it is ```<textarea>```.

![Screenshot from 2025-03-16 03-17-50](https://github.com/user-attachments/assets/a5dd6e81-37bb-448f-85be-d94bac9cba58)


This fix ensures that when a search is performed, DataTables extracts the value of the input field inside the Name column instead of the raw HTML, allowing proper filtering.

### Changes

- Modified the DataTable configuration to properly extract the value from the **input field** in the "Name" column.
- Used ```render``` function to ensure searching works by returning the input field value instead of its HTML content.
- Ensured no visual changes, maintaining the original appearance.

## Changes Reflected

[Screencast from 2025-03-16 03-23-27.webm](https://github.com/user-attachments/assets/ab53dd61-a703-473f-8bef-410c4c8c7417)

## How to test

1. Navigate to **Admin Standard License Comments**.
2. Use the search bar to search for a comment by its name.
3. Verify that results now appear when searching by Name.
4. Ensure searching by Comment content still works as expected.

## Closes
Fixes issue [#3006 ] – "Search by Name Not Working in Admin Standard License Comments"

shaheemazmalmmd commented:
@sakshammishra112 : thank you for your contributions can you check the same issue in acknowledgement page and also in compatibility rules page ??

Please fix it there as well.

sakshammishra112 commented:
## Description
In the **Admin Standard License Comments** section, sorting of **name** column was not working.This issue occurred because of the same reason as why searching was not working.This fix solves both the searching and sorting functionality for both searching and sorting of the **name** column.

## Changes
1. Modified ```t.on('order.dt search.dt', function () {...})``` to correctly renumber rows dynamically.Ensured newly added rows appear at the bottom with the correct serial number.
2. Updated ```$("#addStdLicComment").on('click', function() {...})```to:
   - Append new **rows without disrupting order.**
   - Assign correct **serial numbers dynamically.**
   - Automatically **focus on the new input field**.
   
## Before Modification:

[sortingNotWorking.webm](https://github.com/user-attachments/assets/3e6a9cd3-eced-4188-a928-d830ce20222c)

## After Modification:

[sortingFix.webm](https://github.com/user-attachments/assets/af3a8394-53a8-4865-8e9e-9846217968bf)

## How to test:

1. Open the **Admin License Comment Table**.
2. Try sorting the table by **name column**sand verify that serial numbers update correctly.
3. Search for comments and ensure that the r**ow indexing remains accurate.**
4. Click "**Add Comment**" and confirm that the new row appears at the bottom with the correct number.
5. Modify a comment and submit the form. Check if only modified inputs are updated.

sakshammishra112 commented:
@shaheemazmalmmd 

I have fixed the sorting issue that we discussed earlier on Slack. Also, could you please guide me or provide the link to the acknowledgment and compatibility page? I can't seem to find it.

shaheemazmalmmd commented:
> @shaheemazmalmmd
> 
> I have fixed the sorting issue that i discussed earlier with you on Slack. Also, could you please guide me or provide the link to the acknowledgment and compatibility page? I can't seem to find it.

you could find them here admin > acknowledgments and admin > compatibility rules

shaheemazmalmmd commented:
@sakshammishra112 : We only accept one commit per PR and functionality can you please squash the commit message to one ?

Kaushl2208 commented:
@sakshammishra112 , Any updates on acknowledgement and compatibilty rules????
Can you please take a look and quickly fix for others too?

sakshammishra112 commented:
> @sakshammishra112 , Any updates on acknowledgement and compatibilty rules???? Can you please take a look and quickly fix for others too?

I have fixed the same issue for acknowledgement page, and compatibility rules page works fine

Kaushl2208 commented:
> > @sakshammishra112 , Any updates on acknowledgement and compatibilty rules???? Can you please take a look and quickly fix for others too?
> 
> I have fixed the same issue for acknowledgement page. And compatibility rules page works fine

sakshammishra112 commented:
@Kaushl2208 
How to fix this failing check?

Kaushl2208 commented:
> @Kaushl2208 How to fix this failing check?

[DCO Signing off GitHub Commits](https://cert-manager.io/docs/contributing/sign-off/), Take a look into this :)

Kaushl2208 reviewed (approved):
Changes looks good. Tested. Working as expected for License Acknowledgements, Standard Comments.

Can be merged once the DCO check passes

shaheemazmalmmd reviewed (approved):
Reviewed-By: @Kaushl2208 
Tested-By @Kaushl2208
